tracing/span.go

```diff
@@ -0,0 +1,140 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	otlog "github.com/opentracing/opentracing-go/log"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Span is either an OpenTracing span or an OpenTelemetry span.
+// It could also be a noop span.
+type Span struct {
+	opentracingSpan opentracing.Span
+	otelSpan        trace.Span
+}
+
+// StartSpanFromContext starts a new span from the context using the parent tracing library.
+// If parent span is not sampled, it returns a noop span.
+func StartSpanFromContext(ctx context.Context, operation string, options ...SpanOption) (*Span, context.Context) {
+	otelSpan, opentracingSpan, sampled := SpanFromContext(ctx)
+	if sampled {
```

Comment:
nit: let's do an early return on `!sampled` to reduce indent

Reply:
Done in https://github.com/grafana/dskit/pull/681/commits/80751c9791753d3adf368830cfa28c8bf8d72067